src/Compute/Compute/VirtualMachine/Operation/NewAzureVMCommand.cs

```diff
@@ -510,6 +522,19 @@ public async Task<ResourceConfig<VirtualMachine>> CreateConfigAsync()
                     }
                 }
 
+                //Override Zone logic if PublicIpSku is explicitly provided
+                PublicIPAddressStrategy.Sku publicIpSku;
+                if (_cmdlet.PublicIpSku != null) {
+                    if (_cmdlet.PublicIpSku != "Basic" || _cmdlet.PublicIpSku != "Standard")
```

Comment:
should it be a AND logic instead?

Reply:
OR works too but AND logic makes more sense and sounds correct. Will make the change

Reply:
I do not think OR would work. OR statement make this if statement satisfy no matter what